### PR TITLE
Support for arm64 releases in release-homebrew step

### PR DESCRIPTION
**Pull request by ticky**
This attempts to add arm64 build support to the release-homebrew step

Depends on buildkite/homebrew-buildkite#19, and needs some testing!

### CODE REVIEW

**Comment by yob** [comment]
I think we can test this by changing this ENV var:

https://github.com/buildkite/agent/blob/4c557243c426c210ca1037faca1d407597a9dd6e/.buildkite/pipeline.yml#L2

We might need to change it on master though, and possibly also bump the version to ensure the pipeline triggers DRY_RUN release builds:

https://github.com/buildkite/agent/blob/4c557243c426c210ca1037faca1d407597a9dd6e/agent/version.go#L11

**Comment by yob** [comment]
I think this might be good to go, once we double check the updated formula file in the latest build. Dry run mode is enabled, so we can safely unblock beta and stable releases, then look at the `pkg/buildkite-agent-updated.rb` artifact on the release homebrew step.

**Comment by yob** [comment]
(.. and once this merges, we should probably revert #1349 to a void future confusion)

File: .buildkite/steps/release-homebrew.sh
Comment (ticky) [inline .buildkite/steps/release-homebrew.sh:56]: D’oh!

**Comment by yob** [comment]
The dry run release planned this change, which looks good to me:

```diff
diff --git a/buildkite-agent.rb b/buildkite-agent.rb
index 70bff23..d6571d8 100644
--- a/buildkite-agent.rb
+++ b/buildkite-agent.rb
@@ -3,13 +3,13 @@ class BuildkiteAgent < Formula
   homepage "https://buildkite.com/docs/agent"
 
   stable do
-    version "3.26.0"
-    if Hardware::CPU.physical_cpu_arm64?
-      url     "https://github.com/buildkite/agent/releases/download/v3.26.0/buildkite-agent-darwin-arm64-3.26.0.tar.gz"
-      sha256  "92d731d04912b3de0eb03fe849a0f9eb75554c22ca538a5f388ce3cc66f923e8"
+    version "3.26.1"
+    if Hardware::CPU.arm?
+      url     "https://github.com/buildkite/agent/releases/download/v3.26.1/buildkite-agent-darwin-arm64-3.26.1.tar.gz"
+      sha256  "b5eddcfc02cd2ea29f17ad6aa5bd401fb5042fb35ff01d0af4bf3d5642bf0580"
     else
-      url     "https://github.com/buildkite/agent/releases/download/v3.26.0/buildkite-agent-darwin-amd64-3.26.0.tar.gz"
-      sha256  "8a2f78aad0791ebe41a546a6c17b71560852a439ba311d541d63073410645b2d"
+      url     "https://github.com/buildkite/agent/releases/download/v3.26.1/buildkite-agent-darwin-amd64-3.26.1.tar.gz"
+      sha256  "eec03f649892c92dc4fe779dd37d70cbfca2b199cbad8d5194d6bc1882df7e79"
     end
   end
```

**Review by yob — APPROVED** [review]

**Pull request merged**
